pkg/proc/native/registers_linux_amd64.go

```diff
@@ -324,14 +324,6 @@ func (thread *Thread) fpRegisters() (regs []proc.Register, fpregs proc.LinuxX86X
 	return
 }
 
-type savedRegisters struct {
-	regs   sys.PtraceRegs
-	fpregs proc.LinuxX86Xstate
-}
-
-func (r *Regs) Save() proc.SavedRegisters {
-	savedRegs := &savedRegisters{}
-	savedRegs.regs = *r.regs
-	savedRegs.fpregs = *r.fpregset
-	return savedRegs
+func (r *Regs) Copy() proc.Registers {
```

Comment:
We're just returning the same pointer to the receiver, this breaks the guarantee and we should return an actual copy.

Reply:
It's immutable.

Reply:
Then why have a copy in the first place in the initial `Save()` implementation? Also, how is this immutable, we're returning the receiver pointer directly.

Reply:
> Then why have a copy in the first place
For (1) compatibility with gdbserial where the instance does change and (2) so that in the future we can cache the Regs object

> how is this immutable

The proc.Registers interface doesn't have any methods that mutate the object.

Reply:
Right, but the whole point of `Copy` is that the object won't get modified at all including out from underneath you. Just because the interface cannot mutate doesn't mean anything holding onto the original won't mutate it.

Reply:
I don't think it matters at this stage since we aren't mutating the object from inside proc/native either, but I don't want to fight this so I changed it to make a copy.